.github/workflows/ci.yml

```
@@ -56,6 +56,11 @@ jobs:
           key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
```

```suggestion
          key: ${{ matrix.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
```

This ones the culprit. Using the same cache key for the arm and intel runner. Setting a dedicated key per runner makes it work as expected.

Sample run (only ci.yml updated):
https://github.com/ttytm/wthrr-the-weathercrab/actions/runs/10267829477/job/28409372819

---

It finally works!  Thank you very much for your help, @ttytm! 🎉

It seems as if we would not need to re-install Rust, either.  Feel free to remove that step, if you like.